### PR TITLE
Add support for seccomp actions `ActKillThread` and `ActKillProcess`

### PR DESCRIPTION
**Pull request by saschagrunert**
Two new seccomp actions have been added to the libseccomp-golang
dependency, which can be now supported by runc, too.

ActKillThread kills the thread that violated the rule. It is the same as
ActKill. All other threads from the same thread group will continue to
execute.

ActKillProcess kills the process that violated the rule. All threads in
the thread group are also terminated. This action is only usable when
libseccomp API level 3 or higher is supported.

Xref for requesting a new release in libseccomp-golang: https://github.com/seccomp/libseccomp-golang/issues/55

### CODE REVIEW

**Comment by saschagrunert** [comment]
Removing the WIP since it looks like we do not get a new release of seccomp/libseccomp-golang soon.

File: go.sum [hunk 1]
```diff
@@ -49,6 +49,8 @@ github.com/russross/blackfriday/v2 v2.0.1 h1:lPqVAte+HuHNfhJ/0LC98ESWRz8afy9tM/0
 github.com/russross/blackfriday/v2 v2.0.1/go.mod h1:+Rmxgy9KzJVeS9/2gXHxylqXiyQDYRxCVz55jmeOWTM=
 github.com/seccomp/libseccomp-golang v0.9.1 h1:NJjM5DNFOs0s3kYE1WUOr6G8V97sdt46rlXTMfXGWBo=
 github.com/seccomp/libseccomp-golang v0.9.1/go.mod h1:GbW5+tmTXfcxTToHLXlScSlAvWlF4P2Ca7zGrPiEpWo=
+github.com/seccomp/libseccomp-golang v0.9.2-0.20200616122406-847368b35ebf h1:b0+ZBD3rohnkQ4q5duD1+RyTXTg9yk+qTOPMSQtapO0=
+github.com/seccomp/libseccomp-golang v0.9.2-0.20200616122406-847368b35ebf/go.mod h1:JA8cRccbGaA1s33RQf7Y1+q9gHmZX1yB/z9WDN1C6fg=
```
Comment (kolyshkin) [inline go.sum:53]: Did you forgot to run `go mod tidy`? It should have removed the v0.9.1 checksums (unless they are indirectly used by some other dependency)

Comment (saschagrunert) [reply go.sum:51]: Ah yeah cleaned-up the go.sum file, too.

**Review by AkihiroSuda — APPROVED** [review]

**Comment by cyphar** [comment]
Needs a rebase and probably a small rework given #2750.

**Comment by saschagrunert** [comment]
Rebased on top of the latest master branch.

**Review by cyphar — DISMISSED** [review]
LGTM.

**Comment by AkihiroSuda** [comment]
Needs rebase

**Comment by saschagrunert** [comment]
Rebased on top of the latest `master` branch.

File: go.mod [hunk 2]
```diff
@@ -15,7 +15,7 @@ require (
 	github.com/opencontainers/runtime-spec v1.0.3-0.20210326190908-1c3f411f0417
 	github.com/opencontainers/selinux v1.8.0
 	github.com/pkg/errors v0.9.1
-	github.com/seccomp/libseccomp-golang v0.9.1
+	github.com/seccomp/libseccomp-golang v0.9.2-0.20200616122406-847368b35ebf
```
Comment (AkihiroSuda) [inline go.mod:18]: Not directly related to this PR, but could you include https://github.com/seccomp/libseccomp-golang/commit/3879420cc921efb4f1a2d75bea84e6158ef21a78 (`all: handle EINTR and ENOENT`)

**Review by AkihiroSuda — DISMISSED** [review]

File: go.mod (hunk 2 above)
Comment (saschagrunert) [reply go.mod]: Done, vendored the latest master  :+1:

Comment (AkihiroSuda) [reply go.mod]: thanks

**Review by AkihiroSuda — APPROVED** [review]

**Comment by AkihiroSuda** [comment]
Sorry needs rebase

**Comment by saschagrunert** [comment]
> Sorry needs rebase

Rebased on top of the latest master branch.

**Comment by AkihiroSuda** [comment]
Commit seems broken

**Review by AkihiroSuda — APPROVED** [review]

**Comment by kolyshkin** [comment]
Looks like this will conflict with #2682 and I'd rather have this one rebased _after_ #2682 is merged.

**Comment by kolyshkin** [comment]
#2682 is merged, @saschagrunert can you please rebase?

**Comment by cyphar** [comment]
Given how small the change is, I'll just carry this. I'm sure Sascha won't mind. :wink:

**Comment by cyphar** [comment]
Carried in #3204.

**Comment by saschagrunert** [comment]
Thanks y'all!

**Pull request closed without merging**